lib/ipmi_fru.c

```diff
@@ -4806,14 +4761,14 @@ f_type, uint8_t f_index, char *f_string)
 	else
 	{
 		printf("Wrong field type.");
-		rc = (-1);
+		rc = -1;
 		goto ipmi_fru_set_field_string_out;
 	}
 	memset(fru_data, 0, fru.size);
-	if( read_fru_area(intf ,&fru, fruId, header_offset ,
-					fru_section_len , fru_data) < 0 )
-	{
-		rc = (-1);
+	if (read_fru_area(intf, &fru, fruId, header_offset, fru_section_len,
+			  fru_data)
+	    < 0) {
```

Comment:
This looks weird. Better move the comparison before the function call (`0 > read_fru_area(...`)